prow/sidecar/run.go

```diff
@@ -121,13 +122,44 @@ func (o Options) Run() error {
 		aborted = returnCode == 130
 	}
 
-	return o.doUpload(spec, passed, aborted)
+	metadata := map[string]interface{}{}
+	defer o.doUpload(spec, passed, aborted, metadata)
```

Comment:
What happens to the error from `doUpload` now? This is hard to reason about IMO

Reply:
good point :-)